lib/msf/core/rpc/v10/rpc_module.rb

```diff
@@ -467,6 +467,36 @@ def rpc_options(mtype, mname)
     res
   end
 
+  # Assigns the appropriate dispatcher based on the `mod_type` passed
+  # @param [Msf::Module] mod_type
+  def dispatcher(mod_type)
+    case mod_type
+    when 'encoder'
+      Msf::Ui::Console::CommandDispatcher::Encoder
```

Comment:
In terms of abstraction, I don't think that the RPC interface should call code associated with the user driven console 👀 

It might make sense to extract the payload choosing logic to somewhere else that is more reusable

Reply:
Good call. I'll look into shuffling things around.